Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m

```diff
@@ -60,6 +67,27 @@ + (instancetype)sharedInstance {
     return sharedManager;
 }
 
+-(LPFeatureFlagManager *)featureFlagManager {
+    if (!_featureFlagManager) {
+        _featureFlagManager = [LPFeatureFlagManager sharedManager];
+    }
+    return _featureFlagManager;
+}
+
+-(LPRequestFactory *)requestFactory
+    if (!_requestFactory) {
+        _requestFactory = [[LPRequestFactory alloc] initWithFeatureFlagManager:self.featureFlagManager];
+    }
+    return _requestFactory;
+}
+
+-(LPRequestSender *)requestSender {
```

Comment:
same

Reply:
same

Reply:
If you look deeper at the code, you will notice that it calls sharedInstance. That method itself is thread-safe, making this method idempotent and therefore thread-safe.